### ISSUE TITLE
[BUG] gather list of decimal values now fails

### ISSUE THREAD

**Issue by revans2** [rapidsai/cudf#7441]
**Describe the bug**
The Apache spark plugin started to see test failures in out integration test builds last night https://github.com/NVIDIA/spark-rapids/issues/1802.  I was able to git bisect to find that 2234554c6cfe2941a30401bed907e409bf4bf56c (#7419) caused the issues to show up. I have not traced it down to a specific API yet, but from the tests it appears that gather for an array of decimal values is returning no data, but I am not 100% sure on that. I will try to track it down further.

**Steps/Code to reproduce bug**
Right now it is the spark SQL integration tests, but I will try to come up with a reproducible C++ test. some time soon.  Not sure if #7419 is required or just a nice to have, but this is blocking a lot of Spark plugin work right now.

**Comment by revans2** [comment]
```diff
diff --git a/cpp/tests/copying/gather_list_tests.cu b/cpp/tests/copying/gather_list_tests.cu
index 59f696c440..ac6524b94e 100644
--- a/cpp/tests/copying/gather_list_tests.cu
+++ b/cpp/tests/copying/gather_list_tests.cu
@@ -35,6 +35,7 @@ template <typename T>
 class GatherTestListTyped : public cudf::test::BaseFixture {
 };
 using FixedWidthTypesNotBool = cudf::test::Concat<cudf::test::IntegralTypesNotBool,
+                                                  cudf::test::FixedPointTypes,
                                                   cudf::test::FloatingPointTypes,
                                                   cudf::test::DurationTypes,
                                                   cudf::test::TimestampTypes>;
```
reproduces the problem we are seeing. with results like

```
[ RUN      ] GatherTestListTyped/9.GatherDetailInvalidIndex
../tests/utilities/column_utilities.cu:225: Failure
Failed
depth 2
first difference: lhs[1] = 21474836483, rhs[1] = 3
Google Test trace:
../tests/copying/gather_list_tests.cu:280:  <--  line of failure

[  FAILED  ] GatherTestListTyped/9.GatherDetailInvalidIndex, where TypeParam = numeric::fixed_point<long, (numeric::Radix)10> (1 ms)
[ RUN      ] GatherTestListTyped/9.GatherSliced
../tests/utilities/column_utilities.cu:225: Failure
Failed
depth 2
first difference: lhs[0] = 4, rhs[0] = 7
../tests/utilities/column_utilities.cu:225: Failure
Failed
depth 2
first difference: lhs[0] = 10, rhs[0] = 6
../tests/utilities/column_utilities.cu:225: Failure
Failed
depth 3
first difference: lhs[2] = 30, rhs[2] = 40
../tests/utilities/column_utilities.cu:225: Failure
Failed
depth 3
first difference: lhs[0] = 10, rhs[0] = 0
[  FAILED  ] GatherTestListTyped/9.GatherSliced, where TypeParam = numeric::fixed_point<long, (numeric::Radix)10> (8 ms)
```

I have verified that all of the tests pass if I revert 2234554

**Comment by codereport** [comment]
I have the fix and pushed a PR: https://github.com/rapidsai/cudf/pull/7449. Should be unblocked once this is merged.
